### PR TITLE
Improved and fixed interactive play script

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context

The rendered image and text was flipped the wrong direction. Other sensor images were not being rendered to the screen, now all sensors are automatically rendered. Refactored `observations_to_image` to allow variable sized sensor images. The different image sizes are now tiled side by side. 

## How Has This Been Tested

- Via the interactive play script: `python examples/interactive_play.py --cfg configs/tasks/rearrangepick_replica_cad_example_ik.yaml`
- Added a new test `test_different_dim_observations_to_image` for new changes to `observations_to_image` utility function.
- New test `test_rearrange_example_modules` to make sure the interactive_play script runs. 

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Example of interactive play output:

https://user-images.githubusercontent.com/877440/126211420-e2802555-8d2a-47c1-8ebd-4b6c67aa8bcd.mov

### CODE REVIEW

**Comment by ASzot** [comment]
Example of interactive play output: 
https://user-images.githubusercontent.com/10453381/125808852-fc4e3f07-3eba-4042-a8ef-3232bc7e0272.mov

File: habitat/utils/visualizations/utils.py [hunk 1]
```diff
@@ -192,7 +191,48 @@ def observations_to_image(observation: Dict, info: Dict) -> np.ndarray:
...
+        egocentric_view_l = sorted(
+            egocentric_view_l, key=lambda x: x.shape[0], reverse=True
+        )
+        img_cols = [[egocentric_view_l[0]]]
+        max_height = egocentric_view_l[0].shape[0]
+        cur_y = 0.0
+        # Arrange the images in columns with the largest image to the left.
+        col = []
+        for i in range(len(egocentric_view_l) - 1):
+            im = egocentric_view_l[i + 1]
```
Comment (Skylion007) [inline habitat/utils/visualizations/utils.py:208]:
```suggestion
        for im in egocentric_view_l[1:]:
```

File: habitat/utils/visualizations/utils.py [hunk 2]
```diff
@@ -192,7 +191,48 @@ def observations_to_image(observation: Dict, info: Dict) -> np.ndarray:
...
+            if cur_y + im.shape[0] <= max_height:
+                col.append(im)
+                cur_y += im.shape[0]
+            else:
+                img_cols.append(col)
+                col = [im]
+                cur_y = 0.0
+        img_cols.append(col)
+        col_widths = [
+            max([col_ele.shape[1] for col_ele in col]) for col in img_cols
```
Comment (Skylion007) [inline habitat/utils/visualizations/utils.py:218]:
```suggestion
            max(col_ele.shape[1] for col_ele in col) for col in img_cols
```

File: examples/interactive_play.py [hunk 3]
```diff
@@ -79,15 +79,15 @@ def append_text_to_image(image: np.ndarray, text: str):
     font_size = 0.5
     font_thickness = 1
     font = cv2.FONT_HERSHEY_SIMPLEX
-    blank_image = np.zeros(image.shape, dtype=np.uint8)
+    text_image = np.zeros(image.shape, dtype=np.uint8)
```
Comment (Skylion007) [inline examples/interactive_play.py:82]:
```suggestion
    text_image = np.zeros_like(image)
```
You can specify the dtype if needed

File: examples/interactive_play.py [hunk 4]
```diff
@@ -96,10 +96,7 @@ def append_text_to_image(image: np.ndarray, text: str):
             font_thickness,
             lineType=cv2.LINE_AA,
         )
-    text_image = blank_image[0 : y + 10, 0:w]
-    text_image = cv2.flip(text_image, 0)
-    final = np.concatenate((image, text_image), axis=0)
-    return final
+    return image + text_image
```
Comment (Skylion007) [inline examples/interactive_play.py:99]: What about overflow?

**Review by Skylion007 — DISMISSED** [review]

File: examples/interactive_play.py (hunk 4 above)
Comment (ASzot) [reply examples/interactive_play.py]: Fixed by clipping between 0 and 255.

File: habitat/utils/visualizations/utils.py [hunk 5]
```diff
@@ -164,22 +164,21 @@ def observations_to_image(observation: Dict, info: Dict) -> np.ndarray:
         generated image of a single frame.
     """
     egocentric_view_l: List[np.ndarray] = []
```
Comment (mathfac) [inline habitat/utils/visualizations/utils.py:166]: Looks like it's not egocentric and pretty long non-descriptive name. Maybe rename to `visuals`, `obs_images`, `obs_displays` etc.

File: habitat/utils/visualizations/utils.py [hunk 6]
```diff
@@ -192,7 +191,47 @@ def observations_to_image(observation: Dict, info: Dict) -> np.ndarray:
     assert (
         len(egocentric_view_l) > 0
     ), "Expected at least one visual sensor enabled."
-    egocentric_view = np.concatenate(egocentric_view_l, axis=1)
+
+    shapes = [x.shape for x in egocentric_view_l]
```
Comment (mathfac) [inline habitat/utils/visualizations/utils.py:195]: Let code speak for itself:
```suggestion
    shapes_are_equal = np.all([x.shape for x in egocentric_view_l] == egocentric_view_l[0].shape)
```

File: habitat/utils/visualizations/utils.py [hunk 7]
```diff
@@ -192,7 +191,47 @@ def observations_to_image(observation: Dict, info: Dict) -> np.ndarray:
...
+
+    if len(set(shapes)) != 1:
```
Comment (mathfac) [inline habitat/utils/visualizations/utils.py:197]:
```suggestion
    if shapes_are_equal:
```

File: habitat/utils/visualizations/utils.py [hunk 8]
```diff
@@ -192,7 +191,47 @@ def observations_to_image(observation: Dict, info: Dict) -> np.ndarray:
...
+        # Tile the images, pasting the columns side by side.
+        final_im = np.zeros(
+            (max_height, total_width, 3), dtype=egocentric_view_l[0].dtype
+        )
+        cur_x = 0
+        for i in range(len(img_cols)):
+            next_x = cur_x + col_widths[i]
+            total_col_im = np.concatenate(img_cols[i], axis=0)
+            final_im[: total_col_im.shape[0], cur_x:next_x] = total_col_im
+            cur_x = next_x
```
Comment (mathfac) [inline habitat/utils/visualizations/utils.py:231]: All this code should go in the separate function  `tile_images`.

**Review by mathfac — APPROVED** [review]
Thank you for addressing this. Left the comments mostly related to code readability. Correctness checked using the visual provided.

File: examples/interactive_play.py [hunk 9]
```diff
@@ -262,21 +259,25 @@ def get_wrapped_prop(venv, prop):
...
+    if not args.no_render:
+        obs = env.step({"action": "EMPTY", "action_args": {}})
+        draw_obs = observations_to_image(obs, {})
+        pygame.init()
+        screen = pygame.display.set_mode(
+            [draw_obs.shape[1], draw_obs.shape[0]]
+        )
+
     i = 0
     target_fps = 60.0
```
[the remark this answers is not in the document]
Comment (ASzot) [reply examples/interactive_play.py:282]: Are you referring to `screen`? This is used for rendering.

**Comment by ASzot** [comment]
@mathfac I improved code readability based on comments.

**Pull request merged**